src/api/raw.rs

```diff
@@ -196,12 +196,86 @@ pub struct ListParams {
     pub timeout: Option<u32>
 }
 
-/// Common query parameters for put/post/patch calls
+/// Common query parameters for put/post calls
 #[derive(Default, Clone)]
 pub struct PostParams {
     pub dry_run: bool,
 }
 
+/// Common query parameters for patch calls
+#[derive(Default, Clone)]
+pub struct PatchParams {
+    pub dry_run: bool,
+    /// Strategy which will be used. Defaults to `PatchStrategy::Strategic`
+    pub patch_strategy: PatchStrategy,
+    /// force Apply requests. Applicable only to `PatchStrategy::Apply`
+    pub force: bool,
+    /// fieldManager is a name of the actor that is making changes. Required for `PatchStrategy::Apply`
+    /// optional for everything else
+    pub field_manager: Option<String>
+}
+
+impl PatchParams {
+    fn validate(&self) -> Result<()> {
+        if let Some(field_manager) = &self.field_manager {
+            // that's not complete, go reference code does list all Unicode printable characters as valid, 
+            // this validation is a bit stricter 
+            if field_manager.len() > 128 || !field_manager.chars().all(|f| f.is_alphanumeric() || f == ' ') {
+            return Err(ErrorKind::RequestValidation("Failed to validate PatchParameters::field_manager!".to_owned()).into())
+            }
+        }
+        
+        if self.patch_strategy != PatchStrategy::Apply && self.force {
+             // if not force, all other fields are valid for all types of patch requests
+            Err(ErrorKind::RequestValidation("Force is applicable only for Apply strategy!".to_owned()).into())
+        } else {
+            Ok(())
+        }
+    }
+
+    fn populate_qp(&self, qp: &mut url::form_urlencoded::Serializer<String>) {
+        if self.dry_run {
+            qp.append_pair("dryRun", "true");
+        }
+        if self.force {
+            qp.append_pair("force", "true");
+        }
+        if self.field_manager.is_some() {
+            qp.append_pair("fieldManager", self.field_manager.as_ref().unwrap());
+        }
+    }
+}
+
+
+/// For patch different patch types are supported. See https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment
+/// Apply strategy is kinda special
+#[derive(Clone, PartialEq)]
+pub enum PatchStrategy {
+    Apply,
```

Comment:
wtf is this? i can't even find docs on it

Reply:
https://github.com/kubernetes/apimachinery/blob/7ae370969693753028c74c0c876eee17ddb1b15b/pkg/apis/meta/v1/types.go#L518 
https://github.com/kubernetes/apimachinery/blob/615495ca8cddee7160e9983067a972e8dd572929/pkg/types/patch.go#L24
I thought that i should copy all possible variants, because someone may want this.

Reply:
And just realized that the struct should be renamed to ‘PatchType’

Reply:
Looks like it’s being used in kubectl apply when server supports server side apply: https://github.com/kubernetes/kubernetes/blob/a574bd63ed84ed6a9315d6fddebc7feb7b3242bd/pkg/kubectl/cmd/apply/apply.go#L395
https://github.com/kubernetes/enhancements/issues/555

Reply:
Ahh. Excellent. Thanks!